File: Sakila_Net/Sakila_Net/VB_SakilaLib/DAO/DAOBD/DAOBDFilmActor.vb
'using System;
Imports  System.Data
'using System.Collections.Generic;
'using System.Configuration;
'using System.Collections;
Imports SakilaLib.Cl.Harrys.Sakila.Model
Imports SakilaLib.Cl.Harrys.BD


Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace DAOBD

          Public class DAOFilmActor
             Public  Shared Function SelectAll(nPagina as Integer ,cantidad as Integer) as DataTable
                Dim StSql as String = "Select * from filmActor"
                Return MyBD.getInstance.SqlSelect(StSql, nPagina, cantidad)
        	   End Function

                    Public  Shared Function SelectOne( filmActor as ClFilmActor) as Boolean
             Dim StSql as String = "Select * from filmActor"
            StSql &= " Where  actor_id='" & filmActor.dataActorId.dataActorId & "'" _
                           & "   and film_id='" & filmActor.dataFilmId.dataFilmId & "'" _
                        & ""
                        Dim dt as DataTable = MyBD.getInstance.sqlSelect(stSql)
                        If dt.Rows.Count = 0 Then
                            Return False
                        End If
        filmActor.DataLastUpdate = dt.Rows(0)("last_update").ToString()
                        Return True
                     end Function

        Public  Shared Function SqlInsert( filmActor as ClFilmActor)  as Integer
            Dim stSql as String = "insert into FilmActor (" _
                       & " actor_id ,film_id ,last_update" _
                           & " )values (" _
            & "'" & filmActor.dataActorId.dataactorId & "'" _
                     & ",'" & filmActor.dataFilmId.datafilmId & "'" _
                     & ",'" & filmActor.dataLastUpdate & "'" _
             & ")"
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

        Public  Shared Function SqlDelete( filmActor as ClFilmActor) as Integer
            
            Dim stSql as String =  "delete from FilmActor " _
             & " Where  actor_id='" & filmActor.dataActorId.dataActorId & "'" _
                           & "   and film_id='" & filmActor.dataFilmId.dataFilmId & "'" _
                       
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

       Public   Shared Function SqlUpdate(filmActor as ClFilmActor) as Integer
            Dim stSql as String =  "update FilmActor SET " _
             & " last_update = '" & filmActor.DataLastUpdate & "'" _
             & " Where  actor_id='" & filmActor.dataActorId.dataActorId & "'" _
                           & "   and film_id='" & filmActor.dataFilmId.dataFilmId & "'" _
& ""
            Return MyBD.getInstance.sqlEjecutar(stSql)
            
        end Function
          End Class

         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  

File: Sakila_Net/CS_SakilaLib/DAO/DAOBD/DAOBDFilmActor.cs
using System;
using  System.Data;
using System.Collections.Generic;
using System.Configuration;
using System.Collections;
using Cl.Harrys.Sakila.Model;
using Cl.Harrys.MyBD;


namespace Cl.Harrys.Sakila.DAOBD 
{ 
    public class DAOFilmActor 
    {
        public  static DataTable selectAll(int nPagina  ,int cantidad ) 
        {
            String StSql = "Select  actor_id ,film_id ,last_update  from filmActor";
            return MyBD.MyBD.getInstance().sqlSelect(StSql, nPagina, cantidad);
        }

        public  static Boolean SelectOne( ClFilmActor filmActor)  
        {
            String stSql = "Select  actor_id ,film_id ,last_update  from filmActor";
            stSql += " Where  actor_id='" + filmActor.ActorId.ActorId + "'" 
                           + "   and film_id='" + filmActor.FilmId.FilmId + "'" 
                        + "";
            DataTable dt = MyBD.MyBD.getInstance().sqlSelect(stSql,1,1);
            if (dt.Rows.Count == 0) 
                return false;
                        
        filmActor.LastUpdate = DateTime.ParseExact(dt.Rows[0]["last_update"].ToString(), "yyyy-MM-dd HH:mm:ss,fff",System.Globalization.CultureInfo.InvariantCulture);
            return true;
        }

        public  static int sqlInsert( ClFilmActor filmActor)  
        {
            String stSql = "insert into FilmActor (" 
                       + " actor_id ,film_id ,last_update" 
                           + " )values (" 
            + "'" + filmActor.ActorId.ActorId + "'" 
                     + ",'" + filmActor.FilmId.FilmId + "'" 
                     + ",'" + filmActor.LastUpdate + "'" 
             + ")";
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

        public  static int sqlDelete( ClFilmActor filmActor) {
            
            String stSql  =  "delete from FilmActor " 
             + " Where  actor_id='" + filmActor.ActorId.ActorId + "'" 
                           + "   and film_id='" + filmActor.FilmId.FilmId + "'" 
                       ;
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

       public   static int sqlUpdate(ClFilmActor filmActor ) {
            String stSql  =  "update FilmActor SET " 
             + " last_update = '" + filmActor.LastUpdate + "'" 
             + " Where  actor_id='" + filmActor.ActorId.ActorId + "'" 
                           + "   and film_id='" + filmActor.FilmId.FilmId + "'" 
                 + "";
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
            
        }
   }

}
